qiskit/circuit/quantumcircuit.py

```diff
@@ -1208,7 +1210,7 @@ def qasm(self, formatted=False, filename=None):
             gate._qasm_def_written = False
 
         if filename:
-            with open(filename, 'w+') as file:
+            with codecs.open(filename, 'w+', encoding) as file:
```

Comment:
You don't need to use `codes.open`, the open function supports an encoding kwarg. See: https://docs.python.org/3/library/functions.html#open

so you can just do with `open(filename, 'w+', encoding=encoding) as file:`

Reply:
Thanks again, I changed this too.